src/runtime/device_interface.cpp

```diff
@@ -180,21 +186,30 @@ WEAK int halide_device_sync(void *user_context, struct buffer_t *buf) {
         interface = halide_get_device_interface(buf->dev);
     }
     if (interface == NULL) {
-        return -1;
+        return halide_error_code_no_device_interface;
```

Comment:
Similar checks above also print out a debug message, maybe that would be useful here?